assets/javascripts/discourse/widgets/discourse-reactions-actions.js.es6

```diff
@@ -345,54 +355,69 @@ export default createWidget("discourse-reactions-actions", {
+            : mainReactionIcon
         ).trim();
         const icon = template.content.firstChild;
         icon.style.transform = `scale(${scales[1]})`;
 
         mainReaction.parentNode.replaceChild(icon, mainReaction);
         scaleReactionAnimation(icon, scales[1], scales[0], () => {
           this.dropUserReaction();
-          this.addUserReaction(this.siteSettings.discourse_reactions_like_icon);
+          if (attrs.reaction && attrs.reaction !== mainReactionName) {
+            this.addUserReaction(attrs.reaction);
+          } else {
+            this.addUserReaction(
+              this.siteSettings.discourse_reactions_like_icon
+            );
+          }
 
-          if (!this.attrs.post.current_user_reaction) {
+          if (this.attrs.post.current_user_reaction) {
+            this.attrs.post.reaction_users_count -= 1;
+            this.setCurrentUserReaction();
+            this.attrs.post.current_user_used_main_reaction = false;
+          } else {
             this.attrs.post.reaction_users_count += 1;
             this.setCurrentUserReaction(
               this.siteSettings.discourse_reactions_like_icon
             );
             this.attrs.post.current_user_used_main_reaction = true;
-          } else {
-            this.attrs.post.reaction_users_count -= 1;
-            this.setCurrentUserReaction();
-            this.attrs.post.current_user_used_main_reaction = false;
           }
 
-          CustomReaction.toggle(
-            this.attrs.post.id,
-            this.siteSettings.discourse_reactions_reaction_for_like
-          ).then(value => {
-            resolve();
-            if (value === undefined) {
-              const mainReactionIcon = this.siteSettings
-                .discourse_reactions_like_icon;
-              const hasUsedMainReaction = this.attrs.post
-                .current_user_used_main_reaction;
-              const template = document.createElement("template");
-              template.innerHTML = iconHTML(
-                hasUsedMainReaction
-                  ? `far-${mainReactionIcon}`
-                  : mainReactionIcon
-              ).trim();
-              const mainReaction = template.content.firstChild;
-              icon.parentNode.replaceChild(mainReaction, icon);
-
-              this.attrs.post.current_user_reaction = current_user_reaction;
-              this.attrs.post.current_user_used_main_reaction = current_user_used_main_reaction;
-              this.attrs.post.reactions = reactions;
-              this.attrs.post.reaction_users_count = reaction_users_count;
-              this.scheduleRerender();
+          let toggleReaction =
+            attrs.reaction && attrs.reaction !== mainReactionName
+              ? attrs.reaction
+              : this.siteSettings.discourse_reactions_reaction_for_like;
+
+          CustomReaction.toggle(this.attrs.post.id, toggleReaction).then(
+            value => {
+              resolve();
+
+              if (value === undefined) {
```

Comment:
what are the possible values of value here ?

Reply:
Either Undefined OR json response

Reply:
so why not: `if (!value)` ?

Reply:
when there's the ERROR in API, it returns undefined where `! value` doesn't works

Reply:
<img width="561" alt="Capture d’écran 2021-01-29 à 15 09 11" src="https://user-images.githubusercontent.com/339945/106284844-fe143480-6243-11eb-8185-39e980168e12.png">

I don't undertand

Reply:
I did a complete check for this, it was not needed here. did a change in the new commit.